tests/test_validation.py

```diff
@@ -709,13 +709,13 @@ def validate_hypotest(pdf, data, mu_test, expected_result, tolerance=1e-6):
 @pytest.mark.parametrize(
     'setup_and_tolerance',
     [
-        (setup_1bin_shapesys(), 1e-6),
-        (setup_1bin_lumi(), 4e-6),
-        (setup_1bin_normsys(), 2e-9),
-        (setup_2bin_histosys(), 8e-5),
-        (setup_2bin_2channel(), 1e-6),
-        (setup_2bin_2channel_couplednorm(), 1e-6),
-        (setup_2bin_2channel_coupledhistosys(), 1e-6),
+        (setup_1bin_shapesys(), 4e-5),
+        (setup_1bin_lumi(), 6e-5),
+        (setup_1bin_normsys(), 2e-4),
```

Comment:
Do we know why normsys seems to be so sensitive here? This tolerance is much larger than all the others.

Reply:
Absolutely no idea. Fitted parameters are the same though down to numerical precision.

Reply:
for fixed POI fits I would expect the fitted paramerters to differ

Reply:
ahhhhh, yes, I never checked the fixed poi fits...

Reply:
for 1bin_normsys, we have

master
```
# unconstrained
(Pdb) pyhf.infer.mle.fit(data, pdf).tolist()
[0.30359961840184535, 0.34708322040569417]
# constrained
(Pdb) pyhf.infer.mle.fixed_poi_fit(1.0, data, pdf).tolist()
[1.0, -1.7880486985341282]
```

this branch
```
(Pdb) pyhf.infer.mle.fit(data, pdf).tolist()
[0.30359961840184535, 0.34708322040569417]
(Pdb) pyhf.infer.mle.fixed_poi_fit(1.0, data, pdf).tolist()
[1.0, -1.7880486985341286]
```

Reply:
and the CLs results

master
```
#CLs obs
(Pdb) CLs_obs.tolist()
[0.0006735317010546278]
(Pdb) expected_result['obs']
0.0006735317023683173
(Pdb) (abs(CLs_obs - expected_result['obs']) / expected_result['obs']).tolist()
[1.950449393019978e-09]
# CLs expected
(Pdb) CLs_exp_set.ravel().tolist()
[7.471694617552582e-10, 5.741155155228773e-08, 3.6898088058366224e-06, 0.00016965731536561725, 0.004392708998221363]
(Pdb) expected_result['exp']
[7.47169462e-10, 5.7411551509088054e-08, 3.6898088062731205e-06, 0.00016965731538267896, 0.004392708998555453]
(Pdb) ((CLs_exp_set.ravel() - expected_result['exp'])/expected_result['exp']).tolist()
[-3.275585606661175e-10, 7.52456113642322e-10, -1.1829829768812686e-10, -1.0056572025296775e-10, -7.60556627694298e-11]
```

this branch
```
# CLs obs
(Pdb) CLs_obs.tolist()
[0.000673533629944247]
(Pdb) expected_result['obs']
0.0006735317023683173
(Pdb) (abs(CLs_obs - expected_result['obs']) / expected_result['obs']).tolist()
[2.8618933940777534e-06]

# CLs expected
(Pdb) CLs_exp_set.ravel().tolist()
[7.4725818074954e-10, 5.741738540786033e-08, 3.69012109380142e-06, 0.00016966883322846522, 0.0043929353930099325]
(Pdb) expected_result['exp']
[7.47169462e-10, 5.7411551509088054e-08, 3.6898088062731205e-06, 0.00016965731538267896, 0.004392708998555453]
(Pdb) ((CLs_exp_set.ravel() - expected_result['exp'])/expected_result['exp']).tolist()
[0.0001187397960598631, 0.00010161541743657144, 8.463515176413025e-05, 6.788888389678545e-05, 5.153868707301546e-05]
```

Reply:
To wrap up discussion here --- two tests were added to compare the fitted parameter differences that come out of the minimization routines (optimizers) and ensure they don't change within tolerance. It seems that the CLs has a lot more computations that tiny changes in fitted parameters inflates to larger differences in CLs.